From 2fe22dfdfab8c442a662863fd44826b123abee8f Mon Sep 17 00:00:00 2001
From: Shilin Victor <chrono.monochrome@gmail.com>
Date: Mon, 20 Aug 2018 22:57:43 +0300
Subject: [PATCH 284/296] fix bug that heads up notification appear even when
 disable_expand is set

Test: execute app which setting a disable_expand flag to statusbar.
Test: after that, get a headsup notification.
Test: then, headsup appear and it would be presented as status bar height
---
 .../src/com/android/systemui/statusbar/phone/StatusBar.java        | 7 +++++++
 1 file changed, 7 insertions(+)

diff --git a/packages/SystemUI/src/com/android/systemui/statusbar/phone/StatusBar.java b/packages/SystemUI/src/com/android/systemui/statusbar/phone/StatusBar.java
index 707fbba0660..8786603e68d 100644
--- a/packages/SystemUI/src/com/android/systemui/statusbar/phone/StatusBar.java
+++ b/packages/SystemUI/src/com/android/systemui/statusbar/phone/StatusBar.java
@@ -7764,6 +7764,13 @@ public class StatusBar extends SystemUI implements DemoMode,
             return false;
         }
 
+        if (!panelsEnabled()) {
+            if (DEBUG) {
+                Log.d(TAG, "No peeking: disabled panel : " + sbn.getKey());
+            }
+            return false;
+        }
+
         boolean inUse = mPowerManager.isScreenOn() && !mSystemServicesProxy.isDreaming();
 
         if (!inUse && !isDozing()) {
-- 
2.11.0

